GSA_Adapter/Convert/ToGsa/Group.cs

```diff
@@ -58,12 +58,25 @@ public static partial class Convert
                 return "\"" + load.Objects.Name + "\"";
 
             //Otherwise apply to the corresponding indecies
-            return load.Objects.Elements.Select(x => int.Parse(x.GSAId().ToString())).GeterateIdString();
+            return load.Objects.Elements.SelectMany(x => x.GSAIds()).OrderBy(x => x).GeterateIdString();
 
         }
 
         /***************************************************/
 
+        public static IEnumerable<int> GSAIds(this IBHoMObject obj)
+        {
+            //If FEMesh, then get ID for each face
+            if (obj is FEMesh)
+            {
+                return (obj as FEMesh).Faces.Select(x => x.GSAId());
+            }
+            else
+                return new List<int> { obj.GSAId() };
+        }
```

Comment:
Shouldn't this new public method reside in the Query class and in its own file instead? Ideally with its own description.

Reply:
Sorry didn't realise it was in the Adapter project. Nevermind.

Reply:
oops, should have been private. Will change